### DISCUSSION TITLE
VectorGraphics2D an alternative project with Java2D to vector capabilities

### DISCUSSION THREAD

**Discussion by istinnstudio** [jfree/jfreesvg#50] [Ideas]
https://github.com/eseifert/vectorgraphics2d
This is an open source project I have found with similar Java2D to vector capabilities, could be a useful, alternative source for functions and uses, share code, etc. I report it here as a reference. There is PDF/PS/EPS export support alongside with SVG.

**Comment by jfree** [comment]
Good info, thanks.  If someone has time, perhaps it would be worth running the code through https://github.com/jfree/graphics2d-tester and see what the results look like.

**Comment by mhschmieder** [comment]
I wish I could remember why I rejected it in favor of writing my own EPS in the style of JFreeOrg's SVG and PDF and using those for the other two formats, back in 2019, but unfortunately all of my notes on the subject are lost to the winds due to all the pandemic disruptions in work life.

At first I thought this might have been the one that did direct writes and was super-complex, but it's one of the ones that piggy-backs atop the Graphics 2D API from Core Java. All I know is that I did the evaluation on my own personal time, so my home computer has this 2019 copy of the library. If I do find some notes from my evaluation period, I'll summarize them here.

Taking a quick look at it again tonight, it does seem fairly similar in style to the JFreeOrg approach, making use of Rendering Hints in clever ways as well as making use of a Processing Engine type approach.

The commonalization of EPS, SVG, and PDF handling, is something I was hoping to eventually do at my end as well, which is why I broke out the specific formatting details into a separate sub-package in my own EPS library (before I even saw the Vectorgraphics2D library).

I'm a bit  behind on everything as I had to hop back and forth coast-to-coast three times in 13 months during this crazy time, but as I am once again in a position where going beyond Java 8 isn't yet an option (US Navy contracting firm), I try to stay abreast of all the JFreeOrg activities and progress as well as occasional code reviews, but I haven't had a chance to use the latest versions yet, but it does seem to still be the most actively developed vector graphics export library out there.

**Comment by mhschmieder** [comment]
I'll keep this page bookmarked so that I remember to run the vectorgraphics2D output through your tester if no one else has done so by the time I can come up for air. I get back to open source stuff in spurts, such as tonight, but may not be able to commit to anything until after tax season (mid-April).

It's always good to take fresh look at alternatives every now and then to see if there are areas of unique strength in other options and whether the core JFree libs can architecturally accommodate any improvements that put things on par across-the-board.
